2011-07-29  Ramana Radhakrishnan  <ramana.radhakrishnan@linaro.org>

	gcc/
	Backport only bits marked as [*] from mainline:
        2010-12-19  Chung-Lin Tang  <cltang@codesourcery.com>

	* config/arm/arm.c ([*]arm_legitimate_index_p): Add VFP load/store
	index range case. Change to SF/DFmode tests to avoid capturing HFmode.
	(thumb2_legitimate_index_p): Same.

=== modified file 'gcc/config/arm/arm.c'
--- old/gcc/config/arm/arm.c	2011-06-29 10:46:39 +0000
+++ new/gcc/config/arm/arm.c	2011-07-29 15:01:53 +0000
@@ -5643,8 +5643,8 @@
 
   /* Standard coprocessor addressing modes.  */
   if (TARGET_HARD_FLOAT
-      && (TARGET_FPA || TARGET_MAVERICK)
-      && (GET_MODE_CLASS (mode) == MODE_FLOAT
+      && (TARGET_FPA || TARGET_MAVERICK || TARGET_VFP)
+      && ((mode == SFmode || mode == DFmode)
 	  || (TARGET_MAVERICK && mode == DImode)))
     return (code == CONST_INT && INTVAL (index) < 1024
 	    && INTVAL (index) > -1024

